core/src/services/gcs/backend.rs

```diff
@@ -424,6 +426,10 @@ impl Access for GcsBackend {
 
         m.set_last_modified(parse_datetime_from_rfc3339(&meta.updated)?);
 
+        if let Some(user_metadata) = meta.metadata {
+            m.with_user_metadata(user_metadata);
```

Comment:
should I include a `!user_meta.is_empty()` check before adding the metadata? as done in other services. 

I don't know if gcs can send an empty metadata in the response

Reply:
> should I include a `!user_meta.is_empty()` check before adding the metadata? as done in other services.

I'm thinking about storing `HashMap<String, String>` always to avoid those duplicated checks.

Reply:
done in [#a6478ef](https://github.com/apache/opendal/pull/5276/commits/a6478efa70ac655a3e1f0ae796c79a7eac75537b)